### PR TITLE
fix(logs): use less greedy json regexp

### PR DESCRIPTION
**Pull request by sumo-drosiek**
This change should resolve https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/2843

### CODE REVIEW

File: deploy/helm/sumologic/conf/logs/otelcol/config.yaml [hunk 1]
```diff
@@ -228,7 +228,7 @@ processors:
       - type: json_parser
         parse_from: body
         parse_to: body
-        if: body matches "^{.*"
+        if: body matches "^{.+"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/logs/otelcol/config.yaml:231]:
```suggestion
        if: body matches "^{.*}"
```

File: deploy/helm/sumologic/conf/logs/otelcol/config.yaml [hunk 2]
```diff
@@ -228,7 +228,7 @@ processors:
       - type: json_parser
         parse_from: body
         parse_to: body
-        if: body matches "^{.*"
+        if: body matches "^{.*}"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/logs/otelcol/config.yaml:231]: alternatively it can be:

```suggestion
        if: body matches "^{.+"
```

**Comment by aboguszewski-sumo** [comment]
Does this work with multiline logs?
I did some research and by default `.` doesn't match the newline symbol `\n`.

**Comment by sumo-drosiek** [comment]
seems to be working

**Comment by sumo-drosiek** [comment]
`body matches "^{[\\s\\S]+"` works as expected, but it's pretty hacky

**Comment by sumo-drosiek** [comment]
> > `body matches "^{[\\s\\S]+"` works as expected, but it's pretty hacky
> 
> That's not enough either way, as you also need to handle newlines at the end of the line.

tested on
```
Auto-detected KUBERNETES_SERVICE_DNS_DOMAIN: cluster.local
+ exec /usr/bin/tini -w -e 143 -- java -Dlog4j2.configurationFile=file:/opt/strimzi/custom-config/log4j2.properties -Dvertx.cacheDirBase=/tmp/vertx-cache -Djava.security.egd=file:/dev/./urandom --illegal-access=deny -XX:+ExitOnOutOfMemoryError -XX:MinRAMPercentage=15 -XX:MaxRAMPercentage=20 -classpath lib/io.strimzi.cluster-operator-0.32.0.jar:lib/io.fabric8.openshift-client-6.2.0.jar:lib/io.netty.netty-transport-4.1.77.Final.jar:lib/io.netty.netty-codec-socks-4.1.77.Final.jar:lib/io.fabric8.openshift-model-tuned-6.2.0.jar:lib/io.strimzi.certificate-manager-0.32.0.jar:lib/io.fabric8.kubernetes-model-rbac-6.2.0.jar:lib/io.netty.netty-handler-proxy-4.1.77.Final.jar:lib/io.fabric8.openshift-model-operator-6.2.0.jar:lib/io.fabric8.kubernetes-model-discovery-6.2.0.jar:lib/io.strimzi.kafka-oauth-server-0.11.0.jar:lib/io.fabric8.kubernetes-model-common-6.2.0.jar:lib/io.fabric8.kubernetes-model-storageclass-6.2.0.jar:lib/org.hdrhistogram.HdrHistogram-2.1.11.jar:lib/io.fabric8.kubernetes-model-autoscaling-6.2.0.jar:lib/io.fabric8.kubernetes-model-gatewayapi-6.2.0.jar:lib/com.squareup.okhttp3.logging-interceptor-3.12.12.jar:lib/io.strimzi.config-model-0.32.0.jar:lib/org.apache.zookeeper.zookeeper-jute-3.6.3.jar:lib/io.netty.netty-transport-native-epoll-4.1.77.Final-linux-x86_64.jar:lib/org.apache.zookeeper.zookeeper-3.6.3.jar:lib/io.netty.netty-resolver-dns-4.1.77.Final.jar:lib/net.minidev.accessors-smart-2.4.7.jar:lib/io.netty.netty-handler-4.1.77.Final.jar:lib/com.github.mifmif.generex-1.0.2.jar:lib/io.fabric8.kubernetes-model-certificates-6.2.0.jar:lib/io.netty.netty-codec-http-4.1.77.Final.jar:lib/io.fabric8.openshift-model-miscellaneous-6.2.0.jar:lib/io.strimzi.operator-common-0.32.0.jar:lib/io.fabric8.openshift-model-config-6.2.0.jar:lib/io.fabric8.kubernetes-model-policy-6.2.0.jar:lib/io.strimzi.crd-annotations-0.32.0.jar:lib/com.fasterxml.jackson.core.jackson-annotations-2.13.4.jar:lib/io.fabric8.openshift-model-installer-6.2.0.jar:lib/io.fabric8.openshift-model-storageversionmigrator-6.2.0.jar:lib/io.fabric8.kubernetes-model-node-6.2.0.jar:lib/com.jayway.jsonpath.json-path-2.6.0.jar:lib/net.minidev.json-smart-2.4.7.jar:lib/org.yaml.snakeyaml-1.32.jar:lib/io.fabric8.openshift-model-machine-6.2.0.jar:lib/io.fabric8.zjsonpatch-0.3.0.jar:lib/io.netty.netty-codec-dns-4.1.77.Final.jar:lib/io.fabric8.openshift-model-operatorhub-6.2.0.jar:lib/io.netty.netty-resolver-4.1.77.Final.jar:lib/io.fabric8.openshift-model-6.2.0.jar:lib/io.strimzi.kafka-oauth-server-plain-0.11.0.jar:lib/io.netty.netty-transport-native-unix-common-4.1.77.Final.jar:lib/io.fabric8.openshift-model-console-6.2.0.jar:lib/com.fasterxml.jackson.core.jackson-databind-2.13.4.1.jar:lib/org.quartz-scheduler.quartz-2.3.2.jar:lib/io.prometheus.simpleclient_common-0.7.0.jar:lib/io.fabric8.openshift-client-api-6.2.0.jar:lib/io.netty.netty-transport-native-epoll-4.1.77.Final.jar:lib/org.lz4.lz4-java-1.8.0.jar:lib/com.squareup.okio.okio-1.15.0.jar:lib/org.xerial.snappy.snappy-java-1.1.8.4.jar:lib/io.fabric8.kubernetes-model-apiextensions-6.2.0.jar:lib/org.slf4j.slf4j-api-1.7.36.jar:lib/io.vertx.vertx-core-4.3.4.jar:lib/io.strimzi.kafka-oauth-common-0.11.0.jar:lib/com.fasterxml.jackson.datatype.jackson-datatype-jsr310-2.13.4.jar:lib/io.fabric8.kubernetes-model-events-6.2.0.jar:lib/io.fabric8.openshift-model-whereabouts-6.2.0.jar:lib/io.prometheus.simpleclient-0.7.0.jar:lib/io.fabric8.openshift-model-monitoring-6.2.0.jar:lib/org.apache.yetus.audience-annotations-0.5.0.jar:lib/io.netty.netty-codec-4.1.77.Final.jar:lib/io.fabric8.kubernetes-model-metrics-6.2.0.jar:lib/io.strimzi.api-0.32.0.jar:lib/io.micrometer.micrometer-core-1.3.1.jar:lib/io.fabric8.kubernetes-model-networking-6.2.0.jar:lib/com.nimbusds.nimbus-jose-jwt-9.10.jar:lib/com.squareup.okhttp3.okhttp-3.12.12.jar:lib/org.apache.logging.log4j.log4j-api-2.17.2.jar:lib/org.apache.logging.log4j.log4j-slf4j-impl-2.17.2.jar:lib/io.fabric8.kubernetes-client-6.2.0.jar:lib/com.fasterxml.jackson.dataformat.jackson-dataformat-yaml-2.13.4.jar:lib/com.github.stephenc.jcip.jcip-annotations-1.0-1.jar:lib/io.netty.netty-buffer-4.1.77.Final.jar:lib/io.fabric8.kubernetes-model-flowcontrol-6.2.0.jar:lib/io.fabric8.openshift-model-machineconfig-6.2.0.jar:lib/org.apache.logging.log4j.log4j-core-2.17.2.jar:lib/io.fabric8.openshift-model-hive-6.2.0.jar:lib/org.apache.kafka.kafka-clients-3.3.1.jar:lib/dk.brics.automaton.automaton-1.11-8.jar:lib/com.fasterxml.jackson.core.jackson-core-2.13.4.jar:lib/io.fabric8.kubernetes-client-api-6.2.0.jar:lib/io.netty.netty-codec-http2-4.1.77.Final.jar:lib/io.fabric8.kubernetes-httpclient-okhttp-6.2.0.jar:lib/com.github.luben.zstd-jni-1.5.2-1.jar:lib/io.fabric8.kubernetes-model-admissionregistration-6.2.0.jar:lib/io.fabric8.kubernetes-model-apps-6.2.0.jar:lib/io.fabric8.kubernetes-model-scheduling-6.2.0.jar:lib/io.strimzi.kafka-oauth-client-0.11.0.jar:lib/io.fabric8.kubernetes-model-core-6.2.0.jar:lib/io.netty.netty-transport-classes-epoll-4.1.77.Final.jar:lib/io.vertx.vertx-micrometer-metrics-4.3.4.jar:lib/org.latencyutils.LatencyUtils-2.0.3.jar:lib/io.fabric8.openshift-model-clusterautoscaling-6.2.0.jar:lib/io.fabric8.kubernetes-model-batch-6.2.0.jar:lib/io.micrometer.micrometer-registry-prometheus-1.3.1.jar:lib/io.fabric8.kubernetes-model-extensions-6.2.0.jar:lib/io.fabric8.kubernetes-model-coordination-6.2.0.jar:lib/io.netty.netty-common-4.1.77.Final.jar io.strimzi.operator.cluster.Main
{
  "instant" : {
    "epochSecond" : 1675125364,
    "nanoOfSecond" : 624915000
  },
  "thread" : "main",
  "level" : "INFO",
  "loggerName" : "io.strimzi.operator.cluster.Main",
  "message" : "ClusterOperator 0.32.0 is starting",
  "endOfBatch" : false,
  "loggerFqcn" : "org.apache.logging.log4j.spi.AbstractLogger",
  "threadId" : 1,
  "threadPriority" : 5
}
{
  "instant" : {
    "epochSecond" : 1675125364,
    "nanoOfSecond" : 624915000
  },
  "thread" : "main",
  "level" : "INFO",
  "loggerName" : "io.strimzi.operator.cluster.Main",
  "message" : "ClusterOperator 0.32.0 is starting",
  "endOfBatch" : false,
  "loggerFqcn" : "org.apache.logging.log4j.spi.AbstractLogger",
  "threadId" : 1,
  "threadPriority" : 5
}
```

**Comment by sumo-drosiek** [comment]
https://regex101.com/r/I2YNYe/1

File: deploy/helm/sumologic/conf/logs/otelcol/config.yaml (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/logs/otelcol/config.yaml:231]:
```suggestion
        if: body matches "^{[\\s\\S]+"
```

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**